docs/ref/models/expressions.txt

```diff
@@ -649,6 +666,185 @@ should avoid them if possible.
+normal aggregate functions, however window functions operate on :ref:`frames
+<window-frames>` and partitions, and compute the result for each row, unlike a
+normal aggregation function which computes a final result for each set defined
+by the group by.
+
+It is possible to specify multiple windows in the same query which in Django
+ORM would be equivalent to including multiple expressions in an :doc:`annotate
+</topics/db/aggregation>` method. The ORM does not make use of named windows,
+instead they are part of the selected columns. If you want to apply multiple
+expressions over the same window, e.g., the same partition and frame, it's
+convenient to use a dictionary and pass it as an argument with ``**params``.
+
+.. class:: Window(expression, partition_by=None, order_by=None, frame=None, output_field=None)
+
+.. versionadded:: 2.0
+
+    .. attribute:: filterable
+
+        The SQL standard disallows referencing the window function in the
+        ``WHERE``-clause and Django raises an exception if it's attempted.
+        This scenario may be solved with subqueries. Defaults to ``False``.
+
+    .. attribute:: template
+
+        Defaults to ``%(expression)s OVER (%(window)s)'``. If only the
+        ``expression`` argument is provided, the window clause will be blank.
+
+    The ``Window`` class is the main expression for an ``OVER``-clause. The
+    argument ``expression`` is either a :ref:`window function <window
+    functions>`, an aggregate function or an expression that is compatible in a
+    window clause. The ``partition_by`` controls the partitioning of the
+    rows. Partitioning is used to narrow which rows that are used for the
+    computing the resulting set. The ``partition_by`` argument takes a list of
+    expressions, and thus column names should be wrapped inside an
+    ``F``-object. The ``output_field`` is specified either explicitly, or the
+    ``output_field`` is specified by the expression. The ``order_by``-argument
+    accepts a sequence of expressions on which you can call
+    :meth:`~django.db.models.Expression.asc` and
+    :meth:`~django.db.models.Expression.desc`. The ordering controls in which
+    order the expression is applied, i.e., if you sum over the rows in a
+    partition, the first result is just the value of the first row, the second
+    is the sum of first and second row.
+
+    The ``frame`` parameter specifies which other rows that should be used in
+    the computation, the description of the :ref:`frame-argument
+    <window-frames>` contains more detailed information.
+
+    For example, to add an extra column that has the accumulated average for
+    the movies in the same genre and released in the year. Thus, it's easy to
+    see if a movie is rated better or worse than the average. This is done per
+    movie, and not as a total.
+
+    .. code-block:: python
+
+        >>> from django.db.models import Avg, ExpressionList, F, Window
```

Comment:
`ExpressionList` is imported but not used. Does it need to be a public API? It seems only used internally to `Window`. If it has other uses, could it be added in a separate PR to decrease the size of this one?

Reply:
I'd be happy to make it private API (would it be enough to just remove the docs, and maybe make it an internal class?), and if someone wants it, we can move it out from there. It was a really convenient way to make it more flexible to pass many arguments to `order_by` or `partition`.

Reply:
Removing the docs is enough.